pennylane/circuit_graph.py

```diff
@@ -138,16 +97,28 @@ class CircuitGraph:
     between arbitrary quantum channels and measurements, not just unitary gates.
 
     Args:
-        ops (Iterable[Operator]): quantum operators constituting the circuit, in temporal order
-        variable_deps (dict[int, list[ParameterDependency]]): Free parameters of the quantum circuit.
-            The dictionary key is the parameter index.
-        wires (Wires): the addressable wires on the device
+        ops (Iterable[.Operator]): quantum operators constituting the circuit, in temporal order
+        obs (Iterable[.MeasurementProcess]): terminal measurements, in temporal order
+        wires (.Wires): The addressable wire register of the device that will be executing this graph
+        par_info (dict[int, dict[str, .Operation or int]]): Parameter information. Keys are
+            parameter indices (in the order they appear on the tape), and values are a
+            dictionary containing the corresponding operation and operation parameter index.
+        trainable_params (set[int]): A set containing the indices of parameters that support
+            differentiability. The indices provided match the order of appearence in the
+            quantum circuit.
     """
 
     # pylint: disable=too-many-public-methods
 
-    def __init__(self, ops, variable_deps, wires):
-        self.variable_deps = variable_deps
+    def __init__(self, ops, obs, wires, par_info=None, trainable_params=None):
+        self._operations = ops
```

Comment:
Maybe this makes no sense, but does `ops` include state preparations? The role of state preps confuse me a bit overall tbh.

Reply:
Yep.

Currently the only place where ops and state preps are distinguished are in the queue processing, as there was a suggestion in the tape ADR to eventually distinguish state preps fully from operations.

This stems from issues we've had in the past where state preps don't act like operations - e.g., applying a state prep mid-circuit, trying to take the inverse of a state prep, etc.

However, this change/concept was never finalized across the code base.